examples/mnist_deep_avgpool_mpi.py

```diff
@@ -0,0 +1,276 @@
+        x_image = tf.reshape(x, [-1, 28, 28, 1])
+
+    # First convolutional layer - maps one grayscale image to 32 feature maps.
+    with tf.name_scope('conv1'):
+        W_conv1 = weight_variable([5, 5, 1, 32], "W_conv1")
+        b_conv1 = bias_variable([32])
+        h_conv1 = tf.nn.relu(conv2d(x_image, W_conv1) + b_conv1)
+
+    # Pooling layer - downsamples by 2X.
+    with tf.name_scope('pool1'):
+        h_pool1 = avg_pool_2x2(h_conv1)
+
+    # Second convolutional layer -- maps 32 feature maps to 64.
+    with tf.name_scope('conv2'):
+        W_conv2 = weight_variable([5, 5, 32, 64], "W_conv2")
+        b_conv2 = bias_variable([64])
+        h_conv2 = tf.nn.relu(conv2d(h_pool1, W_conv2) + b_conv2)
+
+    # Second pooling layer.
+    with tf.name_scope('pool2'):
+        h_pool2 = avg_pool_2x2(h_conv2)
+
+    # Fully connected layer 1 -- after 2 round of downsampling, our 28x28 image
+    # is down to 7x7x64 feature maps -- maps this to 1024 features.
+    with tf.name_scope('fc1'):
+        W_fc1 = weight_variable([7 * 7 * 64, 1024], "W_fc1")
+        b_fc1 = bias_variable([1024])
+
+        h_pool2_flat = tf.reshape(h_pool2, [-1, 7 * 7 * 64])
+        h_fc1 = tf.nn.relu(tf.matmul(h_pool2_flat, W_fc1) + b_fc1)
+
+    # Map the 1024 features to 10 classes, one for each digit
+    with tf.name_scope('fc2'):
+        W_fc2 = weight_variable([1024, 10],"W_fc2")
+        b_fc2 = bias_variable([10])
+
+        # y_conv = tf.matmul(h_fc1_drop, W_fc2) + b_fc2
+        y_conv = tf.matmul(h_fc1, W_fc2) + b_fc2
+    return y_conv, tf.placeholder(tf.float32)
+
+
+def conv2d(x, W):
+    """conv2d returns a 2d convolution layer with full stride."""
+    return tf.nn.conv2d(x, W, strides=[1, 1, 1, 1], padding='SAME')
+
+
+def avg_pool_2x2(x):
+    """avg_pool_2x2 downsamples a feature map by 2X."""
+    return tf.nn.avg_pool(
+        x, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
+
+
+def weight_variable(shape, name):
+    """weight_variable generates a weight variable of a given shape."""
+    initial = tf.get_variable(name, shape)
+    return tf.Variable(initial)
+
+
+def bias_variable(shape):
+    """bias_variable generates a bias variable of a given shape."""
+    initial = tf.constant(0.1, shape=shape)
+    return tf.Variable(initial)
+
+def train_mnist_cnn(FLAGS):
+    # Config
+    config = tf.ConfigProto(
+        allow_soft_placement=True,
+        log_device_placement=True,
+        inter_op_parallelism_threads=1)
+
+    # Note: Additional configuration option to boost performance is to set the
+    # following environment for the run:
+    # OMP_NUM_THREADS=44 KMP_AFFINITY=granularity=fine,scatter
+    # The OMP_NUM_THREADS number should correspond to the number of
+    # cores in the system
+
+    # Import data
+    mnist = input_data.read_data_sets(FLAGS.data_dir, one_hot=True)
+
+    xla_device = '/device:NGRAPH:0'
+    if FLAGS.use_xla_cpu:
+        xla_device = '/device:XLA_CPU:0'
+
+    with tf.device(xla_device):
```

Comment:
This needs to be removed as nGraph is not registered as a device.

Reply:
removed